httpx/config.py

```diff
@@ -223,6 +236,43 @@ def __repr__(self) -> str:
         )
 
 
+class HTTPVersionConfig:
```

Comment:
We should probably implement equality and hash for this object to implement the SSLConfig comment.

Reply:
We *might* want contains, iteration, len, equality, and hash on this class. I figure we should probably defer any decisions there vs. this pull request, since it's a bit unclear where we'd want to draw the line.